DirectProgramming/DPC++/DenseLinearAlgebra/guided_jacobi_iterative_gpu_optimization/src/1_guided_jacobi_iterative_solver_cpu.cpp

```diff
@@ -0,0 +1,231 @@
+
+//==============================================================
+// Copyright © 2022 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <bits/stdc++.h>
+
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <ctime>
+#include <oneapi/dpl/random>
+#include <vector>
+
+using namespace sycl;
+
+typedef double real;
+
+// Program variables, feel free to change anything.
+static const int kN = 30000;
+static const real kCheckError = 1e-15;
+static const real kCalculationError = 1e-10;
+static const int kMinRand = -1000;
+static const int kMaxRaand = 1000;
+static const int kMaxSweepps = 100;
+static const std::uint32_t kSeed = 666;
+cpu_selector selector;
+std::ofstream outfile;
+
+// Function responsible for generating a float type
+// diagonally dominant matrix. Float had to be used
+// as using double would result in segmentation faults
+// for extreamlly large matrixes. This is also an example
+// of using sycl based RNG which had to be used as using
+// external (non sycl) functions slows down the execution
+// drasticly.
+void GenerateMatrix(std::vector<float> &input_matrix,
+                    std::vector<real> &input_results) {
+  for (int i = 0; i < kN; ++i) {
```

Comment:
Can you use a more descriptibe name rather then "N" (kN).    This should be what does this represent.   N is a little too "Fortran-like" shorthand.